src/app/components/AllowanceMenu/index.tsx

```diff
@@ -65,13 +99,27 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
     setIsOpen(false);
   }
 
+  function changedPermissionIds(): number[] {
+    if (!permissions || !originalPermissions) return [];
+    const ids = permissions
+      .filter((prm, i) => prm.enabled !== originalPermissions[i].enabled)
+      .map((prm) => prm.id);
+    return ids;
+  }
+
   async function updateAllowance() {
     await msg.request("updateAllowance", {
       id: allowance.id,
       totalBudget: parseInt(budget),
       lnurlAuth,
     });
 
+    await msg.request("disablePermissions", {
```

Comment:
Should we only do this if `changedPermissionIds` is filled?

Reply:
Oops yes, changed!